javers-persistence-mongo/src/main/java/org/javers/repository/mongo/MongoSchemaManager.java

```diff
@@ -41,16 +41,22 @@
         this.mongo = mongo;
     }
 
-    public void ensureSchema() {
+    public void ensureSchema(boolean allowCompoundIndex) {
         //ensures collections and indexes
         MongoCollection<Document> snapshots = snapshotsCollection();
         snapshots.createIndex(new BasicDBObject(GLOBAL_ID_KEY, ASC));
         snapshots.createIndex(new BasicDBObject(GLOBAL_ID_VALUE_OBJECT, ASC));
         snapshots.createIndex(new BasicDBObject(GLOBAL_ID_ENTITY, ASC));
         snapshots.createIndex(new BasicDBObject(GLOBAL_ID_OWNER_ID_ENTITY, ASC));
         snapshots.createIndex(new BasicDBObject(CHANGED_PROPERTIES, ASC));
-        snapshots.createIndex(new BasicDBObject(COMMIT_PROPERTIES + ".key", ASC).append(COMMIT_PROPERTIES + ".value", ASC),
-                new IndexOptions().name(COMMIT_PROPERTIES_INDEX_NAME));
+
+        if (allowCompoundIndex) {
+            snapshots.createIndex(new BasicDBObject(COMMIT_PROPERTIES + ".key", ASC).append(COMMIT_PROPERTIES + ".value", ASC),
+                    new IndexOptions().name(COMMIT_PROPERTIES_INDEX_NAME));
+        } else {
+            snapshots.createIndex(new BasicDBObject(COMMIT_PROPERTIES + ".key", ASC));
```

Comment:
maybe some comment here? // to be removed when DocumentDB adds support to compound indexes